react/src/XMLHTTP/API/pages.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  FavouritesQueryResp,
+  LibraryQueryResp,
+  SearchAllObjectsQueryResp
+} from '@XMLHTTP/types/query'
+import { VERB } from '@cfModule/types/enum'
+import { ToDefine } from '@cfModule/types/common'
+
+/*******************************************************
+ * LIBRARY PAGES
+ *******************************************************/
+
+//Add an outcome to a node
+export function toggleFavourite(
+  objectID,
+  objectType,
+  favourite,
+  callBackFunction = (_data: ToDefine) => console.log('success')
+) {
+  try {
+    $.post(COURSEFLOW_APP.config.post_paths.toggle_favourite, {
```

Comment:
@digrejzo FYI i'm getting rid of these try catches 
on the outer layer 
they are there to protect against passing bad variables in, which should be handled differently 
the try/catch part should be managed by the $.post